### DISCUSSION TITLE
How can I use nested construction with faker?

### DISCUSSION THREAD

**Discussion by artronen** [mockoon/mockoon#1042] [Help] [answered]
Hey there! I want to mock oneOf faked name or NULL
`"myAvatar": "{{ oneOf (array 'faker 'internet.avatar'' 'null' }}"`
but got error

**Comment by 255kb** [comment, marked as answer]
Hi, 

you need to group you helpers with parenthesis, not single quotes: 
`"myAvatar": "{{ oneOf (array (faker 'internet.avatar') 'null') }}"`

With this you will get `null` inside quotes which is not ideal. It's complicated to deal with arrays of mixed values. 

But there is a slightly more complicated solution which will put the quotes only if the value is not null: 

```
{{setVar 'rand' (oneOf (array (faker 'internet.avatar') 'null'))}}
{
  "myAvatar": {{#if (includes @rand 'null') }}{{@rand}}{{else}}"{{@rand}}"{{/if}}
}
```

**Reply by artronen** [reply]
thx!
